hudi-spark/src/main/java/org/apache/hudi/keygen/KeyGenerator.java

```diff
@@ -24,12 +24,17 @@
 import org.apache.avro.generic.GenericRecord;
 
 import java.io.Serializable;
+import java.util.List;
 
 /**
  * Abstract class to extend for plugging in extraction of {@link HoodieKey} from an Avro record.
  */
 public abstract class KeyGenerator implements Serializable {
 
+  protected List<String> recordKeyFields;
```

Comment:
Yes, this was a side effect of merging since the complexKeyGenerator was added explicitly to the test suite, refactored and made this conform the previous abstractions.